can we compile MAX78000 projects with G++ instead of GCC?

I'd like to use C++ code for a project. Changing `main.c` to `main.cpp` in one of the sample projects results in a bunch of `undefined reference to ...` errors.

`make` seems to compile the definition files fine
```
- CC /home/x/MaximSDK/Libraries/MiscDrivers/Camera/camera.c
- CC /home/x/MaximSDK/Libraries/MiscDrivers/Camera/ov7692.c
- CC /home/x/MaximSDK/Libraries/MiscDrivers/Camera/sccb.c
- CC /home/x/MaximSDK/Libraries/MiscDrivers/PMIC/max20303.c
...
```

but i get
```
/home/x/x/x/max78000/main.cpp:334: undefined reference to `camera_start_capture_image()'
```

even though I'm including `camera.h` at the top of `main.cpp`.

The call to `camera_start_capture_image();` is simply that. No arguments passed, no return value.

If I change the name of the definition file `camera.c` (located in MaximSDK files `/home/x/MaximSDK/Libraries/MiscDrivers/Camera`) to `camera.cpp`, the error goes away. But then there is a cascade of new `undefined reference` errors from the files that `camera.cpp` depends on.

Thanks, for what it's worth, these are all the undefined references I get when changing `main.c` to `main.cpp` and attempting to build:

```
/home/x/dev/x/max78000/main.cpp:164: undefined reference to `camera_start_capture_image()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:168: undefined reference to `camera_get_image(unsigned char**, unsigned long*, unsigned long*, unsigned long*)'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:209: undefined reference to `release_camera_stream_buffer()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:176: undefined reference to `get_camera_stream_buffer()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:177: undefined reference to `camera_is_image_rcv()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:216: undefined reference to `get_camera_stream_statistic()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/build/main.o: in function `transmit_capture_uart()':
/home/x/dev/x/max78000/main.cpp:297: undefined reference to `clear_serial_buffer()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:304: undefined reference to `send_msg(char const*)'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:310: undefined reference to `clear_serial_buffer()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:313: undefined reference to `MXC_UART_WriteBytes(mxc_uart_regs_t*, unsigned char const*, int)'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/build/main.o: in function `main':
/home/x/dev/x/max78000/main.cpp:358: undefined reference to `cnn_enable(unsigned long, unsigned long)'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:361: undefined reference to `cnn_boost_enable(mxc_gpio_regs_t*, unsigned long)'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:364: undefined reference to `cnn_init()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:366: undefined reference to `cnn_load_weights()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:368: undefined reference to `cnn_load_bias()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:370: undefined reference to `cnn_configure()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:378: undefined reference to `camera_init(unsigned long)'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:380: undefined reference to `camera_setup(int, int, pixformat_t, fifomode_t, dmamode_t, int)'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:390: undefined reference to `camera_write_reg(unsigned char, unsigned char)'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:422: undefined reference to `cnn_unload(unsigned long*)'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:423: undefined reference to `cnn_stop()'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:426: undefined reference to `softmax_q17p14_q15(long const*, unsigned short, short*)'
/home/x/MaximSDK/Tools/GNUTools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /home/x/dev/x/max78000/main.cpp:413: undefined reference to `cnn_start()'

```

--- a/Libraries/MiscDrivers/Camera/camera.h
+++ b/Libraries/MiscDrivers/Camera/camera.h
@@ -56,6 +56,10 @@
 #include <stdio.h>
 #include <stdint.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #if defined(CAMERA_HM01B0)
 #include "hm01b0_regs.h"
 #elif defined(CAMERA_HM0360_MONO)
@@ -246,5 +250,9 @@ void release_camera_stream_buffer(void);
 
 // Get statistics of DMA streaming mode.
 stream_stat_t *get_camera_stream_statistic(void);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_CAMERA_CAMERA_H_

--- a/Libraries/MiscDrivers/Camera/debayering.h
+++ b/Libraries/MiscDrivers/Camera/debayering.h
@@ -54,6 +54,10 @@
 
 #include <stdint.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 // Return an x coordinate given an array index and x resolution
 unsigned int _x(unsigned int i, unsigned int xres);
 
@@ -118,5 +122,9 @@ void bayer_bilinear_demosaicing(uint8_t *srcimg, uint32_t w, uint32_t h, uint16_
 void bayer_bilinear_demosaicing_crop(uint8_t *srcimg, uint32_t src_width, uint32_t src_height,
                                      uint32_t w_offset, uint32_t h_offset, uint16_t *dstimg,
                                      uint32_t dst_width, uint32_t dst_height);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_CAMERA_DEBAYERING_H_

--- a/Libraries/MiscDrivers/Camera/mipi_camera.h
+++ b/Libraries/MiscDrivers/Camera/mipi_camera.h
@@ -56,6 +56,11 @@
 #ifndef OV5640_DVP
 #include "csi2.h"
 #endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /**
  * @brief Pixel format enumerations.
 */
@@ -243,5 +248,9 @@ mxc_csi2_capture_stats_t mipi_camera_get_capture_stats(void);
  * @return The image header of the last captured image
  */
 char *mipi_camera_get_image_header(void);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_CAMERA_MIPI_CAMERA_H_

--- a/Libraries/MiscDrivers/Camera/sccb.h
+++ b/Libraries/MiscDrivers/Camera/sccb.h
@@ -55,6 +55,10 @@
 #include <stdint.h>
 #include "gpio.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 int sccb_init(void);
 int sccb_vssel(mxc_gpio_vssel_t vssel);
 int sccb_scan(void);
@@ -62,5 +66,9 @@ int sccb_read_byt(uint8_t slv_addr, uint8_t reg, uint8_t *byt);
 int sccb_write_byt(uint8_t slv_addr, uint8_t reg, uint8_t byt);
 int sccb_read_reg16(uint8_t slv_addr, uint16_t reg, uint8_t *byte);
 int sccb_write_reg16(uint8_t slv_addr, uint16_t reg, uint8_t val);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_CAMERA_SCCB_H_

--- a/Libraries/MiscDrivers/Display/display.h
+++ b/Libraries/MiscDrivers/Display/display.h
@@ -52,6 +52,10 @@
 #ifndef LIBRARIES_MISCDRIVERS_DISPLAY_DISPLAY_H_
 #define LIBRARIES_MISCDRIVERS_DISPLAY_DISPLAY_H_
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #define DISP_E_SUCCESS 0
 #define DISP_E_ERROR -1
 #define DISP_E_BUSY -99
@@ -79,5 +83,10 @@ typedef struct {
     uint8_t *comm_buffer;
     uint32_t comm_buffer_len;
 } display_comm_api;
+
+#ifdef __cplusplus
+}
+#endif
+
 
 #endif // LIBRARIES_MISCDRIVERS_DISPLAY_DISPLAY_H_

--- a/Libraries/MiscDrivers/Display/display.h
+++ b/Libraries/MiscDrivers/Display/display.h
@@ -88,5 +88,4 @@ typedef struct {
 }
 #endif
 
-
 #endif // LIBRARIES_MISCDRIVERS_DISPLAY_DISPLAY_H_

--- a/Libraries/MiscDrivers/Display/sharp_mip.h
+++ b/Libraries/MiscDrivers/Display/sharp_mip.h
@@ -57,6 +57,10 @@
 #include "gpio.h"
 #include "display.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /********************************* DEFINES ******************************/
 #define BUF_SIZE(w, h) (((h) * (2 + (w >> 3))) + 2)
 
@@ -82,5 +86,9 @@ void sharp_mip_flush_area(sharp_mip_dev *dev, const display_area_t *area, const 
 void sharp_mip_set_buffer_pixel_util(sharp_mip_dev *dev, uint8_t *buf, uint16_t buf_w, uint16_t x,
                                      uint16_t y, uint8_t color, uint8_t is_opaque);
 void sharp_mip_com_inversion(sharp_mip_dev *dev, int inversion_on);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_DISPLAY_SHARP_MIP_H_

--- a/Libraries/MiscDrivers/Display/ssd1306.h
+++ b/Libraries/MiscDrivers/Display/ssd1306.h
@@ -56,6 +56,10 @@
 #include <stdint.h>
 #include "display.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /********************************* DEFINES ******************************/
 #define BIT_SET(a, b) ((a) |= (1U << (b)))
 #define BIT_CLEAR(a, b) ((a) &= ~(1U << (b)))
@@ -97,5 +101,9 @@ void ssd1306_flush_area(ssd1306_dev *dev, const display_area_t *area, const uint
 
 void ssd1306_set_buffer_pixel_util(uint8_t *buf, uint16_t buf_w, uint32_t buf_max, uint16_t x,
                                    uint16_t y, uint8_t color, uint8_t is_opaque);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_DISPLAY_SSD1306_H_

--- a/Libraries/MiscDrivers/Display/st7735s.h
+++ b/Libraries/MiscDrivers/Display/st7735s.h
@@ -58,6 +58,10 @@
 #ifndef LIBRARIES_MISCDRIVERS_DISPLAY_ST7735S_H_
 #define LIBRARIES_MISCDRIVERS_DISPLAY_ST7735S_H_
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #define DISP_HOR_RES 128
 #define DISP_VER_RES 128
 #define DRAW_BUF_SIZE (DISP_HOR_RES * DISP_VER_RES) / 10
@@ -111,5 +115,9 @@ int st7735s_init(st7735s_cfg_t *cfg);
 int st7735s_pixel(uint32_t x, uint32_t y, uint32_t z);
 int st7735s_write_pixels(uint8_t *data, unsigned int len);
 int st7735s_xyloc(uint8_t row, uint8_t col);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_DISPLAY_ST7735S_H_

--- a/Libraries/MiscDrivers/Display/st7735s_cfaf128128b1.h
+++ b/Libraries/MiscDrivers/Display/st7735s_cfaf128128b1.h
@@ -69,6 +69,10 @@
 
 #include "st7735s.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 st7735s_regcfg_t cfaf128128b1_regcfg[] = {
     /* Send a software reset to clear regs (120ms delay mandatory) */
     { .cmd = ST7735S_SWRESET, .delay = 120, .len = 0, .data = NULL },
@@ -109,5 +113,9 @@ st7735s_regcfg_t cfaf128128b1_regcfg[] = {
     /* Memory Data Access Control; Reverse row and column address order for panel orientation */
     { .cmd = ST7735S_MADCTL, .delay = 0, .len = 1, .data = (uint8_t *)"\xc0" },
 };
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_DISPLAY_ST7735S_CFAF128128B1_H_

--- a/Libraries/MiscDrivers/Display/tft_ili9341.h
+++ b/Libraries/MiscDrivers/Display/tft_ili9341.h
@@ -63,6 +63,10 @@
 #include <gpio.h>
 #include "fonts.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /************************************ DEFINES ********************************/
 #define DISPLAY_WIDTH 320
 #define DISPLAY_HEIGHT 240
@@ -334,5 +338,9 @@ void MXC_TFT_Rectangle(int x0, int y0, int x1, int y1, int color);
  * @param      height       image height
  */
 void MXC_TFT_Stream(int x0, int y0, int width, int height);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_DISPLAY_TFT_ILI9341_H_

--- a/Libraries/MiscDrivers/Display/tft_ssd2119.h
+++ b/Libraries/MiscDrivers/Display/tft_ssd2119.h
@@ -64,6 +64,10 @@
 
 #include "tsc2046.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /************************************************************************************/
 
 typedef struct {
@@ -245,5 +249,9 @@ void MXC_TFT_SetRotation(tft_rotation_t rotation);
  * @param       data                data for the command or register
  */
 void MXC_TFT_WriteReg(uint16_t command, uint16_t data);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_DISPLAY_TFT_SSD2119_H_

--- a/Libraries/MiscDrivers/Display/tft_st7735.h
+++ b/Libraries/MiscDrivers/Display/tft_st7735.h
@@ -63,6 +63,10 @@
 #include <gpio.h>
 #include "fonts.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /************************************ DEFINES ********************************/
 #define DISPLAY_WIDTH 128
 #define DISPLAY_HEIGHT 128
@@ -323,5 +327,9 @@ void MXC_TFT_Line(int x0, int y0, int x1, int y1, int color);
  * @param       color           rectangle color
  */
 void MXC_TFT_Rectangle(int x0, int y0, int x1, int y1, int color);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_DISPLAY_TFT_ST7735_H_

--- a/Libraries/MiscDrivers/Touchscreen/tsc2046.h
+++ b/Libraries/MiscDrivers/Touchscreen/tsc2046.h
@@ -60,6 +60,10 @@
 #include <spi.h>
 #include <gpio.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /************************************************************************************/
 #ifndef TS_MAX_BUTTONS
 #define TS_MAX_BUTTONS 16
@@ -190,5 +194,9 @@ void MXC_TS_RemoveAllButton(void);
  *
  */
 int MXC_TS_GetKey(void);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif // LIBRARIES_MISCDRIVERS_TOUCHSCREEN_TSC2046_H_

